vault/mount.go

```diff
@@ -208,6 +208,11 @@ func (c *Core) mount(entry *MountEntry) error {
 		return logical.CodedError(409, fmt.Sprintf("existing mount at %s", match))
 	}
 
+	// Upgrade to new name
+	if entry.Type == "generic" {
```

Comment:
This and the following change aren't terribly nice to users who are expecting their mounts to look like whatever they're expecting them to look like.

Take a look at credentialAliases in vault/auth.go. That lets people who are still using generic in scripts continue to use generic for both old and new mounts until they're ready to handle both, but will default to using kv for any new mounts. I think we're better off using a similar system here.

Reply:
So that would mean no upgrade case either.  If they are generic now, leave them generic.  I did this originally and forget why I didn't like it.  I will make the appropriate changes.

Reply:
The downside, and maybe why you didn't do it this way originally, is that you can eventually (when you decide to switch over) end up with both generic and kv mounts. But at least people can go in with eyes wide open and on their own time, which I think is better than forcing a change that I know will screw with some peoples' validation logic. :-)